src/sweeper/__main__.py

```diff
@@ -4,60 +4,105 @@
-  sweeper sweep empties thing --workspace c:\\data\\thing
+  sweeper sweep all-checks --workspace=c:\\data\\thing --save-report=c:\\temp --try-fix --backup-to=c:\\temp
 '''
 import os
 import sys
 
 from docopt import docopt
 
-from . import backup, report
+from . import backup, report, workspace_info
 from .sweepers.duplicates import DuplicateTest
 from .sweepers.empties import EmptyTest
 
 
 def main():
-    '''Main entry point for program. Parse arguments and pass to sweeper modules
+    '''Main entry point for program. Parse arguments and pass to sweeper modules.
     '''
-
     args = docopt(__doc__, version='1.0.0')
 
     #: backup input file before quality checks
     if args['--backup-to']:
-        backup.backup_data(os.path.join(args['--workspace'], args['<table_name>']), args['--backup-to'])
+        backup.backup_data(os.path.join(args['--workspace'], args['--table-name']), args['--backup-to'])
 
+    #: check what quality check to run.
     if args['duplicates']:
-        #: do something
-        print(args)
-        # duplicates.find(args['--input'])
-        sweeper = DuplicateTest(args['--workspace'], args['<table_name>'])
+        call_sweeper(args, "DuplicateTest")
     elif args['invalids']:
-        #: do something
-        print(args)
+        call_sweeper(args, "InvalidsTest")
     elif args['empties']:
-        print(args)
+        call_sweeper(args, "EmptyTest")
+    elif args['all-checks']:
+        call_sweeper(args, "DuplicateTest")
+        call_sweeper(args, "EmptyTest")
+
+
+def call_sweeper(args, sweeper_test):
+    '''
+    orchestrate the sweeper calls.
 
-        sweeper = EmptyTest(args['--workspace'], args['<table_name>'])
+    args: the docopt arguments.
+    sweeper_test: the name of the sweeper function to call.
+    '''
+    print(args)
+
+    #: if table name was provided, then run quality check on single table.
+    if args['--table-name']:
+        #: call the quality check.
+        if sweeper_test == "DuplicateTest":
+            sweeper = DuplicateTest(args['--workspace'], args['--table-name'])
+            save_report(args, sweeper, args['--table-name'], 'duplicates')
+        elif sweeper_test == "EmptyTest":
+            sweeper = EmptyTest(args['--workspace'], args['--table-name']) 
+            save_report(args, sweeper, args['--table-name'], 'empties')
+        #: attempt to fix flagged items.
+        if args['--try-fix']:
+            sweeper.try_fix()
+    #: if table name was not provided, then run quality check on all tables in the workpace.
+    else:
+        #: get a list of feature classes that are contained in the workspace.
+        featureclasses = workspace_info.get_featureclasses(args['--workspace'])
+        for fc in featureclasses:
+            print(fc)
+            #: call the quality check.
+            if sweeper_test == "DuplicateTest":
+                sweeper = DuplicateTest(args['--workspace'], fc)
+                save_report(args, sweeper, fc, 'duplicates')
```

Comment:
this is could make a lot of report files. Do you think it would be better to group these by sweeper type and append the logs to it or are you ok with having n * 3 amount of files to review

Reply:
I'm fine either way.  I foresee mainly running sweeper on one table at a time.  and using the 'run on whole workspace' option for the migration process.  that's just me though.   anyone else?  @ZachBeck @eneemann

Reply:
My preference would be one table at a time and if you want to do an entire workspace that can be handled outside of sweeper.

Reply:
Yeah, I agree with Zach

Reply:
ok, so we are removing the run on workspace feature then?

Reply:
@ZachBeck I think he's wondering about how many reports to save to disk if we run an entire workspace.  should it be one report for each quality check or a report for each table/quality check?

Reply:
I'm a bit 🤷‍♂ on this.  In some ways, I think it would be nice to have the reports separate.  That way you could delete all of those with no errors and focus on the remaining reports that have issues noted in them.

Reply:
that is good feedback @eneemann. I can see the usefulness of being able to delete the report after you finish it and move on. Greg and I will work through that tomorrow.

Reply:
Maybe it would be confusing... but what if no errors meant no report? You could just return a print statement that no errors were found.

Reply:
I'd thought about this too.  I don't hate it.  But I also wondered if it would lead to confusion or questions of "where did that report go?"  It seems like it would be an easy change in the code.